storage/tianmu/common/txt_data_format.cpp

```diff
@@ -39,7 +39,9 @@ uint TxtDataFormat::StaticExtrnalSize(ColumnType attrt, uint precision, int scal
   if (attrt == ColumnType::NUM)
     return 1 + precision + (scale ? 1 : 0) + (precision == (uint)scale ? 1 : 0);
   // because "-0.1" may be declared as DEC(1,1), so 18 decimal places may take
-  // 21 characters
+  // 21 characters, "-0.xxxxx...", "-", "0", "." need extra 3 bytes, 21 = 3 + 18.
+  if (attrt == ColumnType::BIT)  // unsigned value.
```

Comment:
This is used by `ColumnType`, and we used int64 to store the bit-value, and as the described by info.

```
//! Number of bytes taken by a value when written out,
//! or number of chars if collation not binary and ColumnType::STRING or ColumnType::VARCHAR
uint TxtDataFormat::StaticExtrnalSize(ColumnType attrt, uint precision, int scale, const DTCollation *col) {
```

should be length of column definition.

Reply:
You are write, the `StaticExtrnalSize` should be column definition ( for bit, the column lenght is `precision`) these two comments is used by  `ColumnType::NUM` and make some confused, I had moved them to the right place above `if (attrt == ColumnType::NUM)`.